tools/mixxx-isolate.py

```diff
@@ -0,0 +1,496 @@
+                except OSError:
+                    pass
+                except Exception as e:
+                    print(e)
+
+        for iso in isolate:
+            with open("/sys/devices/system/cpu/cpu%d/online" % iso, "w") as fp:
+                fp.write("1")
+
+    log.info(" Deactivate cpusets")
+    deleteCpuset(args.cpuset_engine, args.debug)
+    deleteCpuset(args.cpuset_isolation, args.debug)
+    deleteCpuset(args.cpuset_system, args.debug)
+
+    if prev_gov is not None:
+        log.info(" Restore governor %s" % prev_gov)
+        system("sudo cpufreq-set -g %s" % prev_gov)
+
+    log.info(" Cleanup finished")
+    print("STATUS: CLEAN")
+
+
+def createcpuset(name, cpuspec, memspec, cx, mx):
+    try:
+        ccset.create(name, cpuspec, memspec, cx, mx)
+    except CpusetExists:
+        pass
+    except Exception as e:
+        raise e
+
+
+def joinCpuIds(lst):
+    return ",".join((str(x) for x in lst))
+
+
+def isolateSystem(args):
+    conf = selectCpus(args.cpu)
+
+    memspec = "0"
+    log.info(" Create cpuset groups")
+    try:
+        createcpuset(
+            args.cpuset_engine, str(conf["use"]), memspec, False, False
+        )
+        createcpuset(
+            args.cpuset_system,
+            joinCpuIds(conf["system"]),
+            memspec,
+            False,
+            False,
+        )
+        if args.isolate and conf["isolate"]:
+            createcpuset(
+                args.cpuset_isolation,
+                joinCpuIds(conf["isolate"]),
+                memspec,
+                False,
+                False,
+            )
+    except Exception as instance:
+        teardown(args, conf["isolate"])
+        # unroll
+        log.exception(instance)
+        log.critical(" Failed to create shield, hint: do other cpusets exist?")
+        sys.exit(23)
+
+    # move all processes to the system cpuset
+    log.info(" Move processes into system group")
+    root_tasks = cset.unique_set("/").tasks
+    while True:
+        try:
+            proc.move("root", args.cpuset_system, root_tasks, args.debug)
+            break
+        except cset.CpusetException:
+            pass
+
+    if not args.no_offline:
+        for iso in conf["isolate"]:
+            with open("/sys/devices/system/cpu/cpu%d/online" % iso, "w") as fp:
+                fp.write("0")
+
+    # make the task file own by the user, so the engine can add itself later
+    taskFile = (
+        cset.CpuSet.basepath + args.cpuset_engine + cset.CpuSet.tasks_path
+    )
+    os.chown(taskFile, args.uid, -1)
+
+    print("STATUS: ISOLATION")
+
+    def signalHandler(signum, frame):
+        teardown(args, conf["isolate"])
+
+    signal.signal(signal.SIGUSR1, signalHandler)
+
+    while True:
+        try:
+            s = input(">>>")
+            if s == "exit":
+                break
+            else:
+                print("unknown command")
+        except KeyboardInterrupt:
+            break
+        except EOFError:
+            break
```

Comment:
Do we need an interactive prompt? Could we simply wait for `sig` here and not require a TTY?